### DISCUSSION TITLE
How to get access token for a github app installed in a different organization?

### DISCUSSION THREAD

**Discussion by harshtrivedi134** [octokit/octokit.js#2566] [Q&A]
I am currently working on developing a GitHub workflow utilizing Octokit. My goal is to synchronize files from one repository, Repo A (hosted on github.com), to another repository, Repo B (hosted on GitHub Enterprise). These repositories belong to different organizations.

The workflow is running within Repo A, and I am attempting to obtain the access token of the GitHub App installed in the organization of Repo B. My question is: How can I achieve this using the  [app-client](https://github.com/octokit/octokit.js#app-client)?
Thank you in advance for your help!

**Comment by gr2m** [comment]
If you know the repository, you can use the [`GET /repos/{owner}/{repo}/installation`](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-repository-installation-for-the-authenticated-app) endpoint

```js
const { data: installationRepoA } = await app.octokit.request("GET /repos/{owner}/{repo}/installation", { owner: ownerA, repo: repoA })
const { data: installationRepoB } = await app.octokit.request("GET /repos/{owner}/{repo}/installation", { owner: ownerB, repo: repoB })

const installactionOctokitA = await app.getInstallationOctokit(installationRepoA.id)
const installactionOctokitB = await app.getInstallationOctokit(installationRepoA.id)

// use `installactionOctokitA` to read the source file, and `installactionOctokitB` to write the target file
```

**Reply by harshtrivedi134** [reply]
Thanks!  @gr2m 
So basiscaly  the default `app` would only provide pre-authenticated client for current repository & org where the workflow/action is running ?

```
const { App } = require('octokit');
const app = new App({ appId, privateKey });
const octokit = await app.getInstallationOctokit(installationId);
```

**Reply by gr2m** [reply]
no the `app` instance can authenticate as any of its installations, it doesn't matter where the code is run. You can also use the `create-github-app-token` action for this use case if you like: https://github.com/actions/create-github-app-token#create-a-token-for-all-repositories-in-another-owners-installation

**Reply by harshtrivedi134** [reply]
I was trying out the `create-github-app-token` I think it defaults to  the `https://api.github.com/users/{owner}/installation` api. In my case it is github enterprise and the HOSTNAME would be different. I was wondering whether there would be an input to this action that would allow us to override baseUrl but I don't think such attribute exists.

**Reply by gr2m** [reply]
It's not yet implemented, we'd need to add something like this: https://github.com/tibdex/github-app-token/blob/3eb77c7243b85c65e84acfa93fdbac02fb6bd532/action.yml#L8C1-L10C35. I thought we had an open issue for it but doesn't look like it

If you could create an issue and explain your use that'd be fabulous, I can look into it this week

**Reply by harshtrivedi134** [reply]
@gr2m Awesome! I have created an issue - ~~https://github.com/octokit/octokit.js/issues/2569~~ https://github.com/actions/create-github-app-token/issues/77
Looking forward to it!
